core/src/Streamly/Internal/Data/Fold.hs

```diff
@@ -2548,3 +2549,22 @@ top = topBy $ flip compare
 {-# INLINE bottom #-}
 bottom :: (MonadIO m, Storable a, Ord a) => Int -> Fold m a (MA.Array a)
 bottom = topBy compare
+
+{-# INLINE the #-}
+the :: (Eq a, Monad m) => Fold m a (Maybe a)
+the = Fold step initial extract
+
+    where
+
+    initial = do
+        return $ Partial Nothing
+
+    step Nothing x = do
+        return $ Partial (Just x)
+
+    step (Just x0) x = do
+        if x0 == x
+        then return $ Partial (Just x)
```

Comment:
Better use `Just x0` here, or even better to use `step old@(Just x0)` and then use `old` here.